docs/zkapps/o1js/ecdsa.mdx

```diff
@@ -0,0 +1,128 @@
+ECDSA works with different elliptic curves but most commonly with the [secp256k1](/glossary#secp256k1) curve. This curve is also used in Bitcoin and Ethereum.
+
+## Why ECDSA?
+
+To interact with other blockchains and verify data from the outside world, o1js needs to be able to verify signatures.
+ECDSA is a widely used algorithm that is supported by many libraries and tools. For example, Ethereum transactions are signed using ECDSA over the secp256k1 curve.
+As a zkApp developer, when you want to verify an Ethereum transaction and make a statement about it, you must be able to verify the signature of the transaction which is why ECDSA is important for zkApps.
+
+## Basic usage
+
+The ECDSA gadget is used to verify ECDSA signatures. The gadget takes as input the message, the signature, and the public key of the signer. It outputs a `Bool` indicating whether the signature is valid.
+
+Before you can verify a signature, you must initiate the gadget with a curve configuration. 
+
+To initiate the curve:
+
+```ts
+// create a secp256k1 curve
+class Secp256k1 extends createForeignCurve(Crypto.CurveParams.Secp256k1) {}
+```
+
+By default, o1js exports a set of predefined curves. You can use the `createForeignCurve` function to create a curve from a `CurveParams` object. The `CurveParams` object contains the parameters of the curve, such as the modulus, the generator, and the parameters `a` and `b` of the curve equation `y^2 = x^3 + ax + b`.
+
+The namespace `Crypto.CurveParams` exports predefined curves, such as `Pallas`, `Vesta`, and `Secp256k1`.
+
+```ts
+// predefined curve parameters
+CurveParams: {
+  Secp256k1: CurveParams;
+  Pallas: CurveParams;
+  Vesta: CurveParams;
+}
+```
+
+This example uses `Secp256k1` as used in Ethereum. 
+Now that you have a curve, you can create an instance of the ECDSA gadget:
+
+```ts
+// create an instance of ECDSA over secp256k1, previously specified
+class Ecdsa extends createEcdsa(Secp256k1) {}
+```
+
+Before you can verify a signature, you must create one by signing a message. Messages are of type `Bytes`, see [Bytes - API reference](keccak#bytes---api-reference).
+To sign a message, use the `sign` function of the `Ecdsa` class. Note that signing is not a provable operation, only verifying is.
+
+```ts
+// a private key is a random scalar of secp256k1
+let privateKey = Secp256k1.Scalar.random();
+let publicKey = Secp256k1.generator.scale(privateKey);
+
+// create a message, for a detailed explanation of `Bytes` take a look at the Keccak overview
+let message = Bytes32.fromString('cat');
+
+// sign a message - this is not a provable method!
+let signature = Ecdsa.sign(message.toBytes(), privateKey.toBigInt());
+```
+
+Most usages require you to verify a hash of a message instead of the message itself. In Ethereum, for example, the message is the [Keccak](keccak) hash of the transaction data.
```

Comment:
This is a bit misleading, because `sign` and `verify` already include Keccak256 hashing (yes, a bit opinionated). The text makes it sounds like the user might need to hash the message before verifying.

Lower-level methods which sign / verify the EC scalar (which is created from the hash), and thus allow to use any hash function, are available as `signHash()` and `verifySignedHash()`

Reply:
I removed that part, I reference Keccak in a different section so that should be enough!